### PR TITLE
doc: Some doxygen ammendments

### PR DESCRIPTION
**Pull request by OlegHahm**
Mostly exclude some more vendor provided and register defining headers. Also adds a doc-xml target to be used with https://github.com/alobbs/doxy-coverage.

### CODE REVIEW

**Comment by authmillenon** [comment]
:+1:

**Comment by authmillenon** [comment]
Ready to merge?

**Comment by OlegHahm** [comment]
wait a second

**Comment by OlegHahm** [comment]
Amended some more headers. According to doxy-coverage we're at 64% now.

**Comment by kaspar030** [comment]
waitaminute

**Comment by kaspar030** [comment]
IMHO excluding any files that have been authored by us is cheating. That includes, e.g., the encx24j600 headers.

Either we deem documenting all defines necessary, or we don't. Hiding files containing sore spots is just that - hiding it. Doing that for files we've just imported (vendor header files) is fine, but with our own, we should rather live with it, and not "fix" doxygen's exclude path just to get a higher "coverage score".

This is like the proposal that politician did some years back in order to get the unemployment numbers down to the level his government had promised before: let's just not count those that don't want to work.

**Comment by OlegHahm** [comment]
Sorry, Kaspar, but I think your comparison is utter nonsense. And concerning the rest of your post I pretty much disagree, too: We're not going to document registers for CPU or device registers and for a good reason: we are providers of an operating system that support a large amount of hardware, not hardware builders. IMO we're already doing a good job dealing with the often insufficient documentation and driver support from the vendor side, we don't have to make their job wrt documentation, too.

**Comment by thomaseichinger** [comment]
@kaspar030 We're cheating anyway by not counting the unemployed non-citizens, to stay in your analogy. Also `*_regs.h` could be considered as vendor headers. Question is where to draw the border and how to interpret statistics and I agree with @OlegHahm, I am against documenting register defines. They are in the data sheets. I can live with not every single macro documented.

**Comment by OlegHahm** [comment]
Also, there's actually no discussion about cheating or not cheating. I'm not (or not pre-eminently) interested in stats, but in providing a sensible documentation to RIOT users and developers. If someone wants to write code for device X or CPU Y, she has to look in the datasheet of this device/CPU anyway. No reason to include in an OS documentation. I would probably vote for excluding these headers even if they were properly documented. That's just pollution of our API documentation

**Comment by kaspar030** [comment]
I don't want to document the defines. But having the header files (if we wrote them) themselves in doxygen is useful.

**Comment by OlegHahm** [comment]
Ok, that's a whole different discussion as the point you raised in your previous comment. Which headers would you consider useful and why?

**Comment by kaspar030** [comment]
I consider documentation for all header files (excluding the imported vendor files) useful. They contain author information, usually our files have a doxygen "@brief", the documentation can be searched for defines, and doxygen generates an include graph.

**Comment by OlegHahm** [comment]
You consider a file like http://doc.riot-os.org/at86rf2xx__registers_8h.html useful? Seriously?

**Comment by OlegHahm** [comment]
And what about something like http://doc.riot-os.org/native__internal_8h_source.html?

**Comment by kaspar030** [comment]
```http://doc.riot-os.org/native__internal_8h_source.html``` is as is not useful, as the header is missing documentation. But as per current coding convention, that file wouldn't have even been allowed into master.

```http://doc.riot-os.org/at86rf2xx__registers_8h.html``` is a bit lonely without the source that includes it. But it does serve a purpose: document a file (what it is used for, who's the author, general outline of it's content).

**Comment by kaspar030** [comment]
> I would probably vote for excluding these headers even if they were properly documented. That's just pollution of our API documentation

If you think the non-user-visible headers clutter API documentation, it is a simple doxygen flag to just not show the "files" link in the overview, so only those files actually reachable through actual API documentation are reachable.

I find it tremendously useful to have some lines describing what a file is used for and who wrote it. And for consistency, do it with doxygen. This PR proposes to drop that information (by not parsing it and thus not checking for it in the static checks) for files that only contain defines which are not worth documenting.

**Comment by OlegHahm** [comment]
> I find it tremendously useful to have some lines describing what a file is used for and who wrote it. And for consistency, do it with doxygen. This PR proposes to drop that information (by not parsing it and thus not checking for it in the static checks) for files that only contain defines which are not worth documenting.

That's a good summary. I proposed this PR, because I couldn't imagine that anyone could make use of these headers in the documentation. Since there seems to be at least one developer interested in that, I tend to close this.

File: doc/doxygen/riot.doxyfile [hunk 1]
```diff
@@ -815,27 +815,44 @@ EXCLUDE_SYMLINKS       = NO
 # exclude all test directories for example use the pattern */test/*
 
 EXCLUDE_PATTERNS       = */board/*/tools/* \
+                         */boards/*/include/arduino_pinmap.h \
+                         */boards/chronos/drivers/include/display.h \
+                         */boards/*/include/periph_conf.h \
+                         */boards/*/include/board.h \
```
Comment (haukepetersen) [inline doc/doxygen/riot.doxyfile]: why would you exclude the board configuration file? I think it makes very much sense to have them in the documentation!

File: doc/doxygen/riot.doxyfile [hunk 2]
```diff
@@ -815,27 +815,44 @@ EXCLUDE_SYMLINKS       = NO
...
                          */cpu/stm32l1/include/stm32l1xx.h \
+                         */cpu/stm32f0/include/stm32f091xc.h \
                          */cpu/k60/include/MK60D10.h \
                          */cpu/k60/include/MK60DZ10.h \
                          */cpu/kw2x/include/MKW22D5.h \
                          */cpu/k64f/include/MK64F12.h \
                          */cpu/ezr32wg/include/ezr* \
+                         */cpu/*/include/*[_-]regs.h \
+                         */cpu/native/include/native_internal.h \
+                         */drivers/nrf24l01p/include/nrf24l01p_settings.h \
```
Comment (haukepetersen) [inline doc/doxygen/riot.doxyfile]: should not be excluded

File: doc/doxygen/riot.doxyfile [hunk 3]
```diff
@@ -815,27 +815,44 @@ EXCLUDE_SYMLINKS       = NO
...
+                         */drivers/at86rf2xx/include/at86rf2xx_registers.h \
```
Comment (haukepetersen) [inline doc/doxygen/riot.doxyfile]: same here

File: doc/doxygen/riot.doxyfile [hunk 4]
```diff
@@ -815,27 +815,44 @@ EXCLUDE_SYMLINKS       = NO
...
+                         */drivers/*/include/*[_-]internal.h \
+                         */drivers/*/include/*[_-]reg.h \
+                         */drivers/*/include/*[_-]regs.h \
+                         */drivers/encx24j600/include/encx24j600_defines.h \
```
Comment (haukepetersen) [inline doc/doxygen/riot.doxyfile]: same for all of the above.

File: doc/doxygen/riot.doxyfile (hunk 1 above)
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: I couldn't find any interesting documentation in the ones I looked.

File: doc/doxygen/riot.doxyfile (hunk 2 above)
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: Then we should split it up into register definitions and actual settings.

File: doc/doxygen/riot.doxyfile (hunk 3 above)
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: see discussion with @kaspar030.

**Comment by haukepetersen** [comment]
I have to say that I completely agree with @kaspar030 here: Files we wrote ourselfes we should not drop from showing up in doxygen.

File: doc/doxygen/riot.doxyfile (hunk 4 above)
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: I'm still convinced that register definitions and the like should not be included in our API, but if other have other opinions I'm fine with closing this one, too.

**Comment by OlegHahm** [comment]
As I said, fine with me (although I have another opinion), but then we have a lot of documentation to do on this side.

**Pull request closed without merging**